nipype/interfaces/io.py

```diff
@@ -1885,34 +1911,52 @@ class JSONFileSink(IOBase):
     input_spec = JSONFileSinkInputSpec
     output_spec = JSONFileSinkOutputSpec
 
-    def __init__(self, input_names=[], **inputs):
+    def __init__(self, infields=[], force_run=True, **inputs):
         super(JSONFileSink, self).__init__(**inputs)
-        self._input_names = filename_to_list(input_names)
-        add_traits(self.inputs, [name for name in self._input_names])
+        self._input_names = infields
+
+        undefined_traits = {}
+        for key in infields:
+            self.inputs.add_trait(key, traits.Any)
+            self.inputs._outputs[key] = Undefined
+            undefined_traits[key] = Undefined
+        self.inputs.trait_set(trait_change_notify=False, **undefined_traits)
+
+        if force_run:
+            self._always_run = True
+
+    def _process_name(self, name, val):
```

Comment:
the `@` in datasink allows people to dump multiple outputs to the same location and that nipype won't allow connecting two different outputs to the same input. 

i'm not sure yet of all the use cases for this interface, but perhaps we can implement it later if needed.